core/src/test/java/dk/alexandra/fresco/lib/real/LinearAlgebraTest.java

```diff
@@ -0,0 +1,358 @@
+  public void testAdditionUnmatchedHeight() {
+    final int dimension = 10;
+    final int precision = 4;
+
+    Matrix<BigDecimal> matrix1 = allOneMatrix(dimension, dimension, precision);
+    Matrix<BigDecimal> matrix2 = allOneMatrix(dimension - 1, dimension, precision);
+    run((Application<List<Vector<BigDecimal>>, ProtocolBuilderNumeric>) root -> {
+      DRes<Matrix<DRes<SReal>>> closedMatrix = root.realLinAlg().input(matrix1, 1);
+      root.realLinAlg().add(matrix2, closedMatrix);
+      return () -> null;
+    });
+  }
+
+  @Test(expected=IllegalArgumentException.class)
+  public void testAdditionUnmatchedWidth() {
+    final int dimension = 10;
+    final int precision = 4;
+
+    Matrix<BigDecimal> matrix1 = allOneMatrix(dimension, dimension, precision);
+    Matrix<BigDecimal> matrix3 = allOneMatrix(dimension, dimension - 1, precision);
+    run((Application<List<Vector<BigDecimal>>, ProtocolBuilderNumeric>) root -> {
+      DRes<Matrix<DRes<SReal>>> closedMatrix = root.realLinAlg().input(matrix1, 1);
+      root.realLinAlg().add(matrix3, closedMatrix);
+      return () -> null;
+    });
+  }
+
+  @Test(expected=IllegalArgumentException.class)
+  public void testMatrixMultUnmatchedDimensions() {
+    final int dimension = 10;
+    final int precision = 4;
+
+    Matrix<BigDecimal> matrix1 = allOneMatrix(dimension, dimension, precision);
+    Matrix<BigDecimal> matrix2 = allOneMatrix(dimension, dimension - 1, precision);
+    run((Application<List<Vector<BigDecimal>>, ProtocolBuilderNumeric>) root -> {
+      DRes<Matrix<DRes<SReal>>> closedMatrix = root.realLinAlg().input(matrix1, 1);
+      root.realLinAlg().mult(matrix2, closedMatrix);
+      return () -> null;
+    });
+  }
+
+  @Test
+  public void testTransposeMatrix() {
+    // define input and output
+    ArrayList<BigDecimal> rowOne = new ArrayList<>();
+    rowOne.add(BigDecimal.valueOf(1.1));
+    rowOne.add(BigDecimal.valueOf(2.2));
+    ArrayList<BigDecimal> rowTwo = new ArrayList<>();
+    rowTwo.add(BigDecimal.valueOf(3.3));
+    rowTwo.add(BigDecimal.valueOf(4.4));
+    ArrayList<BigDecimal> rowThree = new ArrayList<>();
+    rowThree.add(BigDecimal.valueOf(5.5));
+    rowThree.add(BigDecimal.valueOf(6.6));
+    ArrayList<ArrayList<BigDecimal>> mat = new ArrayList<>();
+    mat.add(rowOne);
+    mat.add(rowTwo);
+    mat.add(rowThree);
+    Matrix<BigDecimal> input = new Matrix<>(3, 2, mat);
+
+    // define functionality to be tested
+    Application<Matrix<BigDecimal>, ProtocolBuilderNumeric> testApplication = root -> {
+      DRes<Matrix<DRes<SReal>>> closed = root.realLinAlg().input(input, 1);
+      DRes<Matrix<DRes<SReal>>> transposed = root.realLinAlg().transpose(closed);
+      DRes<Matrix<DRes<BigDecimal>>> opened = root.realLinAlg().openMatrix(transposed);
+      return () -> new MatrixUtils().unwrapMatrix(opened);
+    };
+    Matrix<BigDecimal> output = run(testApplication);
+    System.out.println(output);
```

Comment:
```suggestion
```

Reply:
Good catch. I removed it in a new commit.